[native] Velox leaks proto headers via dwio

<!--- Provide a general summary of the issue in the Title above -->
<!--- Look through existing open and closed issues to see if someone has reported the issue before -->

## Your Environment
Building presto-native on a new laptop . 

## Expected Behavior
Presto native should build without need protobuf. 

## Current Behavior
```
FAILED: presto_cpp/main/tests/CMakeFiles/presto_server_test.dir/TaskManagerTest.cpp.o 
ccache /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DFMT_LOCALE -DFOLLY_HAVE_INT128_T=1 -DGFLAGS_IS_A_DLL=0 -I/Users/kpai/src/presto/presto-native-execution/. -I/Users/kpai/src/presto/presto-native-execution/velox -I/Users/kpai/src/presto/presto-native-execution/velox/velox/external/xxhash -I/Users/kpai/src/presto/presto-native-execution/_build/debug/velox -I/Users/kpai/src/presto/presto-native-execution/_build/debug -I/Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/xsimd-src/include -I/Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/simdjson-src/simdjson/.. -I/Users/kpai/src/presto/presto-native-execution/_build/debug/presto_cpp/main/thrift -isystem /Users/kpai/homebrew/include -isystem /usr/local/include/antlr4-runtime -isystem /usr/local/include -isystem /Users/kpai/homebrew/opt/openssl@1.1/include -isystem /usr/local/include/proxygen -isystem /Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/gtest-src/googletest/include -isystem /Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/gtest-src/googletest -isystem /Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/gtest-src/googlemock/include -isystem /Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/gtest-src/googlemock -isystem /Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/gtest-src/include -isystem /Users/kpai/src/presto/presto-native-execution/_build/debug/_deps/gtest-src -mcpu=apple-m1+crc -std=c++17 -fvisibility=hidden -fvisibility=hidden -fvisibility-inlines-hidden -Werror -Wno-nullability-completeness -Wno-deprecated-declarations -Wreorder -g -std=gnu++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk -MD -MT presto_cpp/main/tests/CMakeFiles/presto_server_test.dir/TaskManagerTest.cpp.o -MF presto_cpp/main/tests/CMakeFiles/presto_server_test.dir/TaskManagerTest.cpp.o.d -o presto_cpp/main/tests/CMakeFiles/presto_server_test.dir/TaskManagerTest.cpp.o -c /Users/kpai/src/presto/presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp
In file included from /Users/kpai/src/presto/presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp:26:
In file included from /Users/kpai/src/presto/presto-native-execution/velox/velox/dwio/dwrf/reader/DwrfReader.h:19:
In file included from /Users/kpai/src/presto/presto-native-execution/velox/velox/dwio/common/ReaderFactory.h:21:
In file included from /Users/kpai/src/presto/presto-native-execution/velox/velox/dwio/common/BufferedInput.h:20:
In file included from /Users/kpai/src/presto/presto-native-execution/velox/velox/dwio/common/SeekableInputStream.h:23:
/Users/kpai/src/presto/presto-native-execution/velox/velox/dwio/common/wrap/zero-copy-stream-wrapper.h:31:10: fatal error: 'google/protobuf/io/zero_copy_stream.h' file not found
#include <google/protobuf/io/zero_copy_stream.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

```

## Possible Solution
This is because Velox still leaks proto headers via SeekableInputStream.h

## Steps to Reproduce
Make sure that protobuf is not inadvertantly on your include path.

--- a/presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp
+++ b/presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp
@@ -22,11 +22,11 @@
 #include "velox/common/base/Fs.h"
 #include "velox/common/base/tests/GTestUtils.h"
 #include "velox/common/file/FileSystems.h"
+#include "velox/connectors/hive/HiveConnector.h"
+#include "velox/dwio/common/FileSink.h"
+#include "velox/dwio/common/WriterFactory.h"
 #include "velox/dwio/common/tests/utils/BatchMaker.h"
-#include "velox/dwio/dwrf/reader/DwrfReader.h"
-#include "velox/dwio/dwrf/writer/Writer.h"
 #include "velox/exec/Exchange.h"
-#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
 #include "velox/exec/tests/utils/PlanBuilder.h"
 #include "velox/exec/tests/utils/QueryAssertions.h"
 #include "velox/exec/tests/utils/TempDirectoryPath.h"
@@ -39,6 +39,8 @@
 
 DECLARE_int32(old_task_ms);
 DECLARE_bool(velox_memory_leak_check_enabled);
+
+static const std::string kHiveConnectorId = "test-hive";
 
 using namespace facebook::velox;
 
@@ -158,8 +160,7 @@ class TaskManagerTest : public testing::Test {
     auto hiveConnector =
         connector::getConnectorFactory(
             connector::hive::HiveConnectorFactory::kHiveConnectorName)
-            ->newConnector(
-                facebook::velox::exec::test::kHiveConnectorId, nullptr);
+            ->newConnector(kHiveConnectorId, nullptr);
     connector::registerConnector(hiveConnector);
 
     rootPool_ =
@@ -191,8 +192,7 @@ class TaskManagerTest : public testing::Test {
     if (httpServerWrapper_) {
       httpServerWrapper_->stop();
     }
-    connector::unregisterConnector(
-        facebook::velox::exec::test::kHiveConnectorId);
+    connector::unregisterConnector(kHiveConnectorId);
   }
 
   std::vector<RowVectorPtr> makeVectors(int count, int rowsPerVector) {
@@ -213,17 +213,18 @@ class TaskManagerTest : public testing::Test {
   void writeToFile(
       const std::string& filePath,
       const std::vector<RowVectorPtr>& vectors) {
-    dwrf::WriterOptions options;
-    options.config = std::make_shared<facebook::velox::dwrf::Config>();
+    dwio::common::WriterOptions options;
     options.schema = rowType_;
+    options.memoryPool = rootPool_.get();
     auto sink = std::make_unique<dwio::common::LocalFileSink>(
         filePath, dwio::common::FileSink::Options{});
-    dwrf::Writer writer{options, std::move(sink), *rootPool_};
+    auto writer = dwio::common::getWriterFactory(dwio::common::FileFormat::DWRF)
+                      ->createWriter(std::move(sink), options);
 
     for (size_t i = 0; i < vectors.size(); ++i) {
-      writer.write(vectors[i]);
+      writer->write(vectors[i]);
     }
-    writer.close();
+    writer->close();
   }
 
   std::vector<std::shared_ptr<exec::test::TempFilePath>> makeFilePaths(
@@ -247,7 +248,7 @@ class TaskManagerTest : public testing::Test {
     hiveSplit->fileSplit.length = fs::file_size(filePath);
 
     protocol::ScheduledSplit split;
-    split.split.connectorId = facebook::velox::exec::test::kHiveConnectorId;
+    split.split.connectorId = kHiveConnectorId;
     split.split.connectorSplit = hiveSplit;
     split.sequenceId = sequenceId;
     return split;

